policy/modules/system/systemd.if

```diff
@@ -22,6 +22,8 @@ template(`systemd_domain_template',`
     init_nnp_daemon_domain($1_t)
 
     kernel_read_system_state($1_t)
+    selinux_get_enforce_mode($1_t)
+    systemd_read_efivarfs($1_t)
```

Comment:
Are both the lines really required for all systemd services/domains to justify having it in the template? I understand it can simplify updates in the future.

Reply:
As I wrote in the commit message, the code path to read the efivar is in low-level code, so anything that looks at kernel options will hit it. In rawhide:
```console
$ journalctl -b-4 --grep avc.*SystemdOpt -o export|grep _COMM|sort -u
_COMM=systemd-gpt-aut
_COMM=systemd-hostnam
_COMM=systemd-journal
_COMM=systemd-logind
_COMM=systemd-network
_COMM=systemd-resolve
_COMM=systemd-sysctl
_COMM=systemd-timesyn
_COMM=systemd-userdbd
_COMM=systemd-user-ru
_COMM=systemd-userwor
```
So yeah, I think it's reasonable to provide blanket approval.

Reply:
It is still less than half of the systemd domains.
Anyway, if it needed to be allowed for every systemd domain, it would be worth considering to remove `read_efivarfs_type` attribute and allow the permissions to systemd_domain instead - but it currently is just a thought.

```
# seinfo -xasystemd_domain

Type Attributes: 1
   attribute systemd_domain;
        systemd_bootchart_t
        systemd_coredump_t
        systemd_gpt_generator_t
        systemd_hostnamed_t
        systemd_hwdb_t
        systemd_importd_t
        systemd_initctl_t
        systemd_localed_t
        systemd_logger_t
        systemd_logind_t
        systemd_machined_t
        systemd_modules_load_t
        systemd_networkd_t
        systemd_notify_t
        systemd_passwd_agent_t
        systemd_resolved_t
        systemd_rfkill_t
        systemd_sysctl_t
        systemd_timedated_t
        systemd_tmpfiles_t
        systemd_userdbd_t

# seinfo -xasystemd_read_efivarfs_type

Type Attributes: 1
   attribute systemd_read_efivarfs_type;
        systemd_gpt_generator_t
        systemd_hostnamed_t
        systemd_hwdb_t
        systemd_localed_t
        systemd_logind_t
        systemd_machined_t
        systemd_modules_load_t
        systemd_networkd_t
        systemd_resolved_t
        systemd_rfkill_t
        systemd_sysctl_t
        systemd_timedated_t
        systemd_userdbd_t
```